resources-for-container/files/env-runner/backend_services_data_importer.py

```diff
@@ -6,15 +8,15 @@ class BackendServicesDataImporter:
 
     def __init__(self, env: Environment):
         self.env = env
+        self.test_data_dump_filepath: str = self.env.mount_directory + "/test_data.sql"
+
+        self._check_test_data_file_is_present()
```

Comment:
Why are we putting this logic here? Is there some way we could make this self-documenting and avoid the need for the explanatory comment in `start.py`?

Reply:
I was wondering the same thing.
There are mainly two positive points about the current choice:
- it makes sense for the check to be part of the constructor, as the file being present is really a prerequisite for this class to be able to do its job.
- by putting the check in the constructor, I also avoid the need to do the same check inside [BackendServicesDataImporter::populate_postgres_with_test_data()](https://github.com/alphagov/digitalmarketplace-contained/blob/test-data-alert/resources-for-container/files/env-runner/backend_services_data_importer.py#L15), as it was before (which I think it is a good thing to do)

On the other hand is not too bad to remove the call to `check_test_data_file_is_present()` from the constructor and call it explicitly after the object is constructed (that was actually the first version I did).

What would you do?

Reply:
That all sounds sensible. Why do we need to know about this feature in `start.py`? Could we just remove that comment? I don't generally find comments about what code is doing very valuable.

And relatedly, I agree that it makes sense for the check to be here in the constructor. We want to reduce the gap between setting a thing and checking it is valid to reduce the risk of future bugs. Could we go one step further and do the check before we assign the variable? Something like `self.test_data_dump_filepath = self._get_test_data_dump_filepath()`, perhaps?

Reply:
@bjgill and I had a chat, and we agreed this change is a good compromise https://github.com/alphagov/digitalmarketplace-contained/pull/27/commits/d4470406558f3b42c6790502584b4ac65d223bb2

Reply:
@bjgill and I had a chat, and we agreed this change is a good compromise https://github.com/alphagov/digitalmarketplace-contained/pull/27/commits/d4470406558f3b42c6790502584b4ac65d223bb2

Reply:
I've implemented your suggestion to do `self.test_data_dump_filepath = self._get_test_data_dump_filepath()` https://github.com/alphagov/digitalmarketplace-contained/pull/27/commits/cd3f6ce028bfa7fbabf0f08872047b94c93af643